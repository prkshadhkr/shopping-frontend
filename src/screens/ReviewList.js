import React from 'react';
import ReviewInfo from './ReviewInfo'

const ReviewList = ({ reviews = [], deleteReview }) => {
  if(reviews.length < 1) {
    return (
    <h6 style={{textAlign:"center"}}>
      No review exists yet!
    </h6>)
  }
  
  return (
    reviews.map(r => (
      <ReviewInfo 
        key={r.id}
        id={r.id}
        title={r.title}
        rating={r.rating}
        comment={r.comment}
        deleteReview={deleteReview}
      />
    ))
  );
}

export default ReviewList;