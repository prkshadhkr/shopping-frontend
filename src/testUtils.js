import React from  'react';
import UserContext from './user/UserContext';

/** following is for mock store */
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import axios from 'axios'
import MockAdapter from 'axios-mock-adapter';

const mock = new MockAdapter(axios);
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialState = {};
const makeMockStore = (state = {}) => {
  return mockStore ({
    ...initialState,
    ...state
  })
}

/** create a demouser */
const demoUser = {
  username: "testing",
  first_name: "testing",
  last_name: "testing",
  email: "testing@gmail.com",
  password: "testing",
  is_admin: false
};

/**
 * user is destructured from userContext in Profilejs as
 * const { user, setUser } = useContext(UserContext);
 */
const UserProvider = ({ children, user = demoUser }) => (
  <UserContext.Provider value={{user}}>
    {children}
  </UserContext.Provider>
)

export { mock, makeMockStore, UserProvider };