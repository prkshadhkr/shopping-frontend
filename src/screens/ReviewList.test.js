import React from 'react';
import { render } from '@testing-library/react';
import ReviewList from './ReviewList';
import { MemoryRouter } from 'react-router';
import { Provider } from "react-redux";
import { makeMockStore } from '../testUtils';

const store = makeMockStore();

it ('renders without crashing', function() {
  render(
    <MemoryRouter>
      <Provider store={store}>
        <ReviewList />
      </Provider>
    </MemoryRouter>
  )
});

it ('matches snapshot', function() {
  const { asFragment } = render (
    <MemoryRouter>
      <Provider store={store}>
        <ReviewList />
      </Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
})