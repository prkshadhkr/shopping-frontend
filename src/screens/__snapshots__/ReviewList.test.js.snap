// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h6
    style="text-align: center;"
  >
    No review exists yet!
  </h6>
</DocumentFragment>
`;
